src/RecommendedOptions.cmake

```diff
@@ -0,0 +1,130 @@
+# ENABLE_DEVELOPER_MODE: sets defaults appropriate for developers, this is defaulted to ON
+#  * WARNINGS_AS_ERRORS: ON
+#  * ENABLE_SANITIZER_ADDRESS: ON
+#  * ENABLE_CLANG_TIDY: ON for unix
+#  * ENABLE_SANITIZER_UNDEFINED: ON for unix
+#  * ENABLE_CPPCHECK: ON for unix
+
+# For non-developer builds
+#  -DENABLE_DEVELOPER_MODE:BOOL=OFF
+# Is recommended
+
+# In developer mode, all features have options that show up in the CMake GUI tools
+
+# recommended_project_options() macro enables all recommended defaults with appropriately
+# applied options from the GUI which are set
+
+# Any default can be overridden
+# set(<feature_name>_DEFAULT <value>) - set default for both user and developer modes
+# set(<feature_name>_DEVELOPER_DEFAULT <value>) - set default for developer mode
+# set(<feature_name>_USER_DEFAULT <value>) - set default for user mode
+
+option(ENABLE_DEVELOPER_MODE "Set up defaults for a developer of the project, and let developer change options" ON)
+
+if(MSVC)
```

Comment:
How would this behave when using `clang` on Windows? Should this be `if(WIN32)`?

Reply:
I believe these checks are correct. If you're not using MSVC, you're using a different generator, (Make or Ninja likely) and these things should all work as expected.

The bigger question I had, is, should this include all of the non-Make/Ninja generators, since those do not support clang-tidy et al.

Reply:
https://cmake.org/cmake/help/latest/variable/MSVC.html?highlight=msvc
Suggests that's not the case. Especially, if one is using Ninja generator on Windows with environment variables `CC=cl.exe` and `CXX=cl.exe`

I am with you that this should not include generators that don't support `clang-tidy`.

Reply:
OK thanks, I'll update this to be more explicit based on generator, not compiler.

Reply:
I have made the variable names and checks much more explicit